SimPEG/electromagnetics/natural_source/receivers.py

```diff
@@ -70,14 +74,20 @@ def _locs_e(self):
         if self.locations.ndim == 3:
             loc = self.locations[:, :, 0]
         else:
-            loc = self.locations
+            if self.locations.shape[1] == 6:
```

Comment:
Just looking back on this, I am not entirely sure why we would end up with `locations` being 6 wide... is this if the e / b receivers are in different locations? If that is the case, I think it might be better to instead have keywords for the e-locations, and b-locations?

Reply:
@lheagy this was a hack for that case exactly :) Some of the data I work with is roving B and stationary E. I can certainly take the time to add the keywords. I think it would be worth given the rise in airborne AMT techniques.

Reply:
:+1: And we also don't have an explicit base station either, which might be interesting to have the possibility for. But for now, we can think through the `locations` assignment. My initial thought would be to have it similar to how we instantiate the dipole receivers. maybe: 

```
locations=None, locations_e=None, locations_b=None
```

and then the user can set either of `locations` (which then assumes e & b are co-located) or `locations_e`, `locations_b` which can be distinct

Reply:
Sounds great. I like it. I've also had plans of evaluating the E receiver as dipole and this plays right into that. But sometime down the road.

Reply:
`locations` does not assume that e and b are co-located. It allows for the E and B fields to be in separate locations with an additional dimension added to the matrix ([nr_rx, 3, 2] shape (dim=3)). Then the e_loc = locations[:,:,0] and b_loc = locations[:,:,1] and can be assigned independently of each other. This isn't a well-documented feature of the receiver class but I support the idea of having keyword arguments @lheagy. 
I do also think adding better mechanisms for the locations would be great, as @JKutt mention above, especially when using a dipole or not-aligning with grid data.